File: selected_coding_interview/codes/python/lc_231_power_of_two.py
"""
File: lc_231_power_of_two.py
Created Time: 2025-12-30
Author: krahets
"""

from include import *


# ===== Solution Code =====
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and n & (n - 1) == 0


# ======= Test Case =======
# Test case 1
# TODO: Add appropriate test case

# ====== Driver Code ======
slt = Solution()
# result = slt.isPowerOfTwo(...)
# print(result)

File: selected_coding_interview/codes/java/lc_231_power_of_two/lc_231_power_of_two.java
/*
* File: lc_231_power_of_two.java
* Created Time: 2025-12-30
* Author: krahets
*/

package lc_231_power_of_two;

import include.*;
import java.util.*;

// ===== Solution Code =====
    class Solution {
        public boolean isPowerOfTwo(int n) {
            return n > 0 && (n & (n - 1)) == 0;
        }
    }

public class lc_231_power_of_two {
    public static void main(String[] args) {
        // ======= Test Case =======
        // Test case 1
        // TODO: Add appropriate test case

        // ====== Driver Code ======
        Solution slt = new Solution();
        // result = slt.isPowerOfTwo(...)
        // print(result)

    }
}
